adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala

```diff
@@ -121,6 +121,121 @@ trait GenomicRDD[T, U <: GenomicRDD[T, U]] {
     replaceRdd(tFn(rdd))
   }
 
+  private[rdd] val sortedTrait: SortedTrait
+  private[rdd] def sorted = sortedTrait.sorted
+  private[rdd] def partitionMap = sortedTrait.partitionMap
+
+  /**
+   * This case class is the source of all sorted knowledge handling in ADAM. We
+   * have it as an inner class currently because there were some access issues
+   */
+  private[rdd] case class SortedTrait(sorted: Boolean,
+                                      partitionMap: Option[Seq[(ReferenceRegion, ReferenceRegion)]]) extends Serializable
+  /**
+   * This repartition method repartitions all data in this.rdd and distributes it as evenly as possible
+   * into the number of partitions provided. The data in this.rdd is already sorted (by ReferenceRegion.start)
+   * so we ensure that it is kept that way as we move data around.
+   *
+   * Data is repartitioned in lists to maintain the sorted order. After the repartition, the lists are placed
+   * in the correct order based on their original sorted order.
+   *
+   * @param partitions the number of partitions to repartition this.rdd into
+   * @return a new SortedGenomicRDDMixIn with the RDD partitioned evenly
+   */
+  private def evenlyRepartition(partitions: Int)(implicit tTag: ClassTag[T]): GenomicRDD[T, U] = {
+    assert(sorted)
+    // the average number of records on each node will help us evenly repartition
+    val average: Double = rdd.count.toDouble / partitions
+    // we already have a sorted rdd, so let's just move the data
+    // but we want to move it in blocks to maintain order
+    // zipWithIndex seems to be the cheapest way to guarantee this
+    val finalPartitionedRDD = rdd
+      .mapPartitionsWithIndex((idx, iter) => {
+        // here we get the destination partition number
+        // this will calculate so that each partition has a nearly
+        // equal amount of data on it
+        inferCorrectReferenceRegionsForPartition(iter, partitionMap.get(idx))
```

Comment:
It is a much better name than "assumeIncorrectReferenceRegionsForPartition", IMNSHO.